Civi/Api4/Service/Spec/SpecFormatter.php

```diff
@@ -307,10 +307,12 @@ public static function setInputTypeAndAttrs(FieldSpec $fieldSpec, $data, $dataTy
       self::setLegacyDateFormat($inputAttrs);
     }
     // Number input for numeric fields
-    if ($inputType === 'Text' && in_array($dataTypeName, ['Integer', 'Float'], TRUE)) {
+    if ($inputType === 'Text' && in_array($dataTypeName, ['Integer', 'Float', 'Money'], TRUE)) {
       $inputType = 'Number';
+    }
+    if ($inputType === 'Number') {
       // Todo: make 'step' configurable for the custom field
-      $inputAttrs['step'] = $dataTypeName === 'Integer' ? 1 : .01;
+      $inputAttrs['step'] ??= $dataTypeName === 'Integer' ? 1 : .01;
```

Comment:
This looks good, except that not all currencies have 2 significant digits. For that matter, not all floats do either. So let's change this to
```suggestion
      $inputAttrs['step'] ??= $dataTypeName === 'Integer' ? 1 : 'any';
```